src/Microsoft.AspNetCore.Sockets.Abstractions/Features/IConnectionInherentKeepAliveFeature.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.AspNetCore.Sockets.Features
+{
+    /// <summary>
+    /// Indicates if the connection transport has an "inherent keep-alive", which means that the transport will automatically
+    /// inform the client that it is still present.
+    /// </summary>
+    /// <remarks>
+    /// The most common example of this feature is the Long Polling HTTP transport, which must (due to HTTP limitations) terminate
+    /// each poll within a particular interval and return a signal indicating "the server is still here, but there is no data yet".
+    /// This feature allows applications to add keep-alive functionality, but limit it only to transports that don't have some kind
+    /// of inherent keep-alive.
+    /// </remarks>
+    public interface IConnectionInherentKeepAliveFeature
+    {
+        TimeSpan KeepAliveInterval { get; }
```

Comment:
Does this property provide any value? It isn't used currently

Reply:
It exposes the duration of the inherent keep-alive (i.e. for Long Polling, the poll timeout value). To be honest, this is just used as a marker interface though, so we could drop it. It seemed like a useful value to have though.

Reply:
It would cleanup the LongPolling constructor a little :P

Reply:
But marker interfaces are groooooooss

Reply:
I'm with @BrennanConroy, I don't think this adds any value. Maybe this interface should have a bool so that keep alives can be manually turned off.